agent/plugin/plugin.go

```diff
@@ -289,28 +288,26 @@ func (p *Plugin) constructRepositoryHost() (string, error) {
 	if len(parts) < 2 {
 		return "", fmt.Errorf("Incomplete plugin path \"%s\"", p.Location)
 	}
-
+	
 	var s string
+	switch parts[0] {
+		case "github.com", "bitbucket.org":
+			s = strings.Join(parts[:3], "/")
```

Comment:
I think in all of these switch branches, it'll be clearer if we have
```suggestion
			return strings.Join(parts[:3], "/"), nil
```

rather than assigning to a variable that we return at the bottom. Minimizing cognitive load for readers and all that